chapter04/GamblersProblem.py

```diff
@@ -20,9 +20,13 @@
 # optimal policy
 policy = np.zeros(GOAL + 1)
 
-# state value
+# reward Value
+rewardValue = np.zeros(GOAL + 1)
+rewardValue[GOAL] = 1.0
+
+# state Value
 stateValue = np.zeros(GOAL + 1)
-stateValue[GOAL] = 1.0
```

Comment:
Exercise 4.9 only talks about the trick so we don't need to have another array for rewards.
It's asynchronous because there is only one state value array. So the value iteration is in-place. See section 4.5 for more details.